File: DataBindingBasicFJava/app/src/main/java/com/websarva/wings/android/databindingbasicfjava/MainFragment.java
package com.websarva.wings.android.databindingbasicfjava;

import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProvider;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.websarva.wings.android.databindingbasicfjava.databinding.FragmentMainBinding;
import com.websarva.wings.android.databindingbasicfjava.ui.MainViewModel;

public class MainFragment extends Fragment {
	private MainViewModel _mainViewModel;
	private FragmentMainBinding _fragmentMainBinding;
	@Override
	public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
		_fragmentMainBinding = FragmentMainBinding.inflate(inflater, container, false);
		View contentView = _fragmentMainBinding.getRoot();
		return contentView;
	}

	@Override
	public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
		super.onViewCreated(view, savedInstanceState);

		ViewModelProvider provider = new ViewModelProvider(MainFragment.this);
		_mainViewModel = provider.get(MainViewModel.class);

		_fragmentMainBinding.setRandNum(_mainViewModel.getRandNumStr());
		_fragmentMainBinding.btnRand.setOnClickListener(new ButtonClickListener());
	}

	@Override
	public void onDestroyView() {
		super.onDestroyView();
		_fragmentMainBinding = null;
	}

	private class ButtonClickListener implements View.OnClickListener {
		@Override
		public void onClick(View v) {
			_mainViewModel.generateNewRandNum();
			_fragmentMainBinding.setRandNum(_mainViewModel.getRandNumStr());
		}
	}
}

File: DataBindingBasicFKotlin/app/src/main/java/com/websarva/wings/android/databindingbasicfkotlin/MainFragment.kt
package com.websarva.wings.android.databindingbasicfkotlin

import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.viewModels
import com.websarva.wings.android.databindingbasicfkotlin.databinding.FragmentMainBinding
import com.websarva.wings.android.databindingbasicfkotlin.ui.MainViewModel

class MainFragment : Fragment() {
	private val _mainViewModel by viewModels<MainViewModel>()
	private var _fragmentMainBindingInit: FragmentMainBinding? = null
	private val _fragmentMainBinding get() = _fragmentMainBindingInit!!

	override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
		_fragmentMainBindingInit = FragmentMainBinding.inflate(inflater, container, false)
		return _fragmentMainBinding.root
	}

	override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
		super.onViewCreated(view, savedInstanceState)
		_fragmentMainBinding.randNum = _mainViewModel.getRandNumStr()
		_fragmentMainBinding.btnRand.setOnClickListener(ButtonClickListener())
	}

	override fun onDestroyView() {
		super.onDestroyView()
		_fragmentMainBindingInit = null;
	}

	private inner class ButtonClickListener : View.OnClickListener {
		override fun onClick(view: View?) {
			_mainViewModel.generateNewRandNum()
			_fragmentMainBinding.randNum = _mainViewModel.getRandNumStr()
		}
	}
}
